public static List<SObject> create(List<SObject> records, int batchSize) throws UnexpectedErrorFault, InvalidIdFault, LoginFault, RemoteException, IllegalArgumentException, Exception {
    logger.log(Level.INFO, "Ready to insert: " + records.size() + " records.");
    List<SObject> returnList = new ArrayList<SObject>();
    SaveResult[] saveResults = null;
    boolean errorFound = false;
    String errors = new String();
    SObject[] objs = new SObject[records.size()];
    try {
        if (batchSize > dmlStatementSize || batchSize < 1)
            throw new IllegalArgumentException("batchSize must be between 1 and 200");
        for (int j = 0; j < records.size(); j++) objs[j] = records.get(j);
        saveResults = binding.create(objs, mySessionHeader, null, null, null, null, null, null);
        // Handle the results
        for (int i = 0; i < saveResults.length; i++) {
            // Determine whether create succeeded or had errors
            if (saveResults[i].getSuccess()) {
                // No errors, so we will retrieve the id created for this index
                records.get(i).setId(saveResults[i].getId());
                returnList.add(records.get(i));
                logger.log(Level.INFO, "Item " + i + " id" + saveResults[i].getId());
            } else {
                // Handle the errors
                errorFound = true;
                logger.log(Level.SEVERE, "ClientSalesforce.create/Item " + i + " had an error creating.");
                logger.log(Level.SEVERE, "ClientSalesforce.create/The error reported was: " + saveResults[i].getErrors()[0].getMessage() + "\n");
                errors += saveResults[i].getErrors()[0].getMessage() + "\n";
            }
        }
        if (errorFound)
            throw new Exception("errors found: " + errors);
        logger.log(Level.INFO, "Query executed sucessfull. Returned Values: " + records.size());
    } catch (RemoteException e) {
        logger.log(Level.SEVERE, "ClientSalesforce.create/Error: " + e);
    } catch (InvalidIdFault e) {
        logger.log(Level.SEVERE, "ClientSalesforce.create/Error: " + e);
    } catch (UnexpectedErrorFault e) {
        logger.log(Level.SEVERE, "ClientSalesforce.create/Error: " + e);
    } catch (Exception e) {
        logger.log(Level.SEVERE, "ClientSalesforce.create/Error: " + e);
    }
    return returnList;
}
